### ISSUE TITLE
[commitment] move evaluation proof creation to its own module

### ISSUE THREAD

**Issue by mimoo** [o1-labs/proof-systems#384]
A polynomial commitment is three algorithms:

1. commit
2. open (or as I like to call it, create an evaluation proof)
3. verify

These three algorithms are all specified in the same file which makes it hard to parse. We should move the opening stuff in its own module under the same crate. Specifically, [OpeningProof](https://github.com/o1-labs/proof-systems/blob/master/poly-commitment/src/commitment.rs#L228) and the [open](https://github.com/o1-labs/proof-systems/blob/master/poly-commitment/src/commitment.rs#L709) function should be moved to `evaluation_proof.rs` or something.

**Comment by xiangxiecrypto** [comment]
Good starting point to contribute, will try.

**Comment by mimoo** [comment]
https://github.com/o1-labs/proof-systems/pull/397
